src/v/cluster/node_status_provider.cc

```diff
@@ -119,6 +121,8 @@ void node_status_provider::tick() {
 }
 
 ss::future<> node_status_provider::dispatch_node_status_requests() {
+    if (!_feature_table.local().is_active(feature::raftless_node_status)) {
+        return ss::make_ready_future<>();
```

Comment:
i'd recommend sequencing the commit which adds the feature towards the beginning of the PR and then squashing these changes into the commit which adds the node status provider.

Reply:
Yeah. I've re-ordered the commits. Should make more sense now.